### ISSUE TITLE
feature request: start consuming with given offset

### ISSUE THREAD

**Issue by tabVersion** [ditsuke/google-pubsub-demo#5]
In the demo, I don't see the related part. 

Please consider the case: the program pulls messages from the cloud and records the offset for each message. When it crashes and recovers, we can continue to consume messages from where it has stopped.

**Comment by ditsuke** [comment]
Question: for messages we have already consumed and acked, rolling back the offset will require us to set a (billable) retention policy for the subscription. Is that acceptable?

Beyond this I'll need to investigate whether the google-cloud-pubsub crate supports seeking yet.

**Comment by tabVersion** [comment]
> rolling back the offset will require us to set a (billable) retention policy for the subscription. Is that acceptable?

So pubsub won't keep every message for a period of time, eg. 24h. What if we don't ack message? Can you summarize the billing policy? 

> Beyond this I'll need to investigate whether the google-cloud-pubsub crate supports seeking yet.

I recommend referring to google's pubsub API doc rather than the crate's doc

**Comment by ditsuke** [comment]
> So pubsub won't keep every message for a period of time, eg. 24h.

Not by default, however we can configure a retention policy on _subscriptions_ for up to 7 days.
[pubsub/docs#subscription_message_retention][1]

> What if we don't ack message?

If we don't ack the message, it'll be redelivered once a subscription-specific `ack_deadline` exceeds.
This is configurable. Defaults to 10s, limited to 10 minutes.
[pubsub/docs/.../api/modifyAckDeadline][2]

> Can you summarize the billing policy?

Billing is based on storage allocated to retained, acknowledged messages (free for 7 days for unacknowledged messages). Retaining acknowledged messages with a subscription-based retention policy will bill the subscription's project for storage costs.
[pubsub/pricing#storage_costs][3]
 
> I recommend referring to google's pubsub API doc rather than the crate's doc

Noted. The API supports settings retention duration on both topics and subscriptions. Assuming topic configuration is outside RisingWave's purview, we'll be able to set a retention policy of up to 7 days on subscriptions to make seeking possible subject to storage costs.

[1]: https://cloud.google.com/pubsub/docs/replay-overview#subscription_message_retention
[2]: https://cloud.google.com/pubsub/docs/reference/rest/v1/projects.subscriptions/modifyAckDeadline
[3]: https://cloud.google.com/pubsub/pricing#storage_costs
